node/network/bridge/src/network.rs

```diff
@@ -303,3 +303,17 @@ impl Network for Arc<NetworkService<Block, Hash>> {
 		);
 	}
 }
+
+/// We assume one peer_id per authority_id.
+pub async fn get_peer_id_by_authority_id<AD: AuthorityDiscovery>(
```

Comment:
So this would start to break if many nodes changed their PeerIds at once, without rotating their session keys.
I believe this is acceptable, as honest nodes will practically always rotate their keys before changing the machine / VM their node is running on, due to the risk of equivocation / slashing.

Reply:
If a node changes their `PeerId`, I'd assume they restart the node and connect under the new `PeerId` and disconnect under the old one. They'd also publish a new record on the DHT with their new `PeerId`. It's true that our cache might contain the old record when that happens.
But the list of our neighbors is not necessarily connected to us, we'll use random peers if there are not enough.